src/ast/passes/semantic_analyser.cpp

```diff
@@ -1722,6 +1722,27 @@ void SemanticAnalyser::validate_map_key(const SizedType &key, Node &node)
 
 void SemanticAnalyser::visit(Map &map)
 {
+  auto initializer = dynamic_cast<MapInitializer *>(map.initializer);
+  if (initializer) {
+    auto found = BPF_MAP_TYPES.find(initializer->bpf_type);
+    if (found == BPF_MAP_TYPES.end()) {
+      auto &err = initializer->addError();
+      err << "Invalid bpf map type: " << initializer->bpf_type;
+      std::stringstream hint;
```

Comment:
The hint is a stream, intentionally to avoid this boilerplate, remember? :)  Just do:
```
auto &hint = err.addHint();
hint << "Valid map types: ";
....
   hint << bpf_type.first;
   if (num_types != 0) {
      hint << ", ";
   }
```

And eliminate the `err.addHint()` line below.

Reply:
I forget so quickly! Will fix.